chain/chain/src/flat_storage_resharder.rs

```diff
@@ -0,0 +1,1008 @@
+            error!(target: "resharding", "flat storage resharding of {col_name} is not implemented yet!");
+        }
+        _ => unreachable!(),
+    }
+    Ok(())
+}
+
+/// Performs post-processing of shard splitting after all key-values have been moved from parent to children.
+/// `success` indicates whether or not the previous phase was successful.
+fn split_shard_task_postprocessing(
+    resharder: FlatStorageResharderInner,
+    task_status: FlatStorageReshardingTaskStatus,
+) {
+    let (parent_shard, split_status) = get_parent_shard_and_status(&resharder);
+    let SplittingParentStatus { left_child_shard, right_child_shard, flat_head, .. } = split_status;
+    let flat_store = resharder.runtime.store().flat_store();
+    info!(target: "resharding", ?parent_shard, ?task_status, ?split_status, "flat storage shard split task: post-processing");
+
+    let mut store_update = flat_store.store_update();
+    match task_status {
+        FlatStorageReshardingTaskStatus::Successful => {
+            // Split shard completed successfully.
+            // Parent flat storage can be deleted from the FlatStoreManager.
+            resharder
+                .runtime
+                .get_flat_storage_manager()
+                .remove_flat_storage_for_shard(parent_shard, &mut store_update)
+                .unwrap();
+            store_update.remove_flat_storage(parent_shard);
+            // Children must perform catchup.
+            for child_shard in [left_child_shard, right_child_shard] {
+                store_update.set_flat_storage_status(
+                    child_shard,
+                    FlatStorageStatus::Resharding(FlatStorageReshardingStatus::CatchingUp(
+                        flat_head.hash,
+                    )),
+                );
+            }
+            // TODO(trisfald): trigger catchup
+        }
+        FlatStorageReshardingTaskStatus::Failed | FlatStorageReshardingTaskStatus::Interrupted => {
+            // We got an error or an interrupt request.
+            // Reset parent.
+            store_update.set_flat_storage_status(
+                parent_shard,
+                FlatStorageStatus::Ready(FlatStorageReadyStatus { flat_head }),
+            );
+            // Remove children shards leftovers.
+            for child_shard in [left_child_shard, right_child_shard] {
+                store_update.remove_flat_storage(child_shard);
+            }
+        }
+    }
+    store_update.commit().unwrap();
+    // Terminate the resharding event.
+    *resharder.resharding_event.lock().unwrap() = None;
+}
+
+/// Copies a key-value pair to the correct child shard by matching the account-id to the provided shard layout.
+fn copy_kv_to_child(
+    status: &SplittingParentStatus,
+    key: Vec<u8>,
+    value: FlatStateValue,
+    store_update: &mut FlatStoreUpdateAdapter,
+    account_id_parser: impl FnOnce(&[u8]) -> Result<AccountId, std::io::Error>,
+) -> Result<(), std::io::Error> {
```

Comment:
unsure why are we using std::io::Error instead of a more concrete type like resharding error? If we really do want to use a generic error type, we can go for anyhow::Error here instead.

Reply:
I used the error of `parse_account_id_` functions, but I'm fine to use resharding error